### ISSUE TITLE
Dev - user can type in composer while 'Delete request' modal is opened

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#25585]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open an existing chat
2. Request money
3. Click on the IOU preview component & navigate to the IOU thread report
4. Hover over the IOU preview component
5. Click on the 'Delete' icon
6. Start typing
## Expected Result: 
Text shouldn't be entered to the composer
## Actual Result: 
Text is entered to the composer

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** DEV
**Reproducible in staging?:** DEV
**Reproducible in production?:** DEV
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/2f18f439-c7ec-4e2f-9cab-65c35569e84b

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Natnael-Guchima
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692623660379109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
This is not a blocker for now

**Comment by danieldoglas** [comment]
Agreed its a dupe of https://github.com/Expensify/App/issues/24045, closing this one out.

**Comment by Natnael-Guchima** [comment]
@danieldoglas this is issue is only reproducible on Dev, and the other one is reproducible on production. Would that make the issues different? 
Thanks.

**Comment by danieldoglas** [comment]
@Natnael-Guchima both are opening a `ConfirmModal`, which is the root cause. Once we fix the focus issue for that modal, both issues will be solved. That's why it's the same issue.

**Comment by Natnael-Guchima** [comment]
Copied that! Thanks 👍
